packages/odata-v2/src/de-serializers/temporal-de-serializers.ts

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable valid-jsdoc */
+import { Temporal } from '@js-temporal/polyfill';
+import { DeSerializer } from '@sap-cloud-sdk/odata-common/internal';
+
+/**
+ * Temporal (de-)serializers for Odata-v2.
+ * @internal
+ */
+export const temporalDeSerializers = {
+  'Edm.DateTime': {
+    deserialize: deserializeToPlainDateTime,
+    serialize: serializeFromPlainDateTime,
+    serializeToUri: (value, serialize) =>
+      `datetime'${serialize(value).toString().replace(/Z$/, '')}'`
+  } as DeSerializer<Temporal.PlainDateTime>,
+  'Edm.DateTimeOffset': {
+    deserialize: deserializeToZonedDateTime,
+    serialize: serializeFromZonedDateTime,
+    serializeToUri: (value, serialize) => `datetimeoffset'${serialize(value)}'`
+  } as DeSerializer<Temporal.ZonedDateTime>,
+  'Edm.Time': {
+    deserialize: deserializeToPlainTime,
+    serialize: serializeFromPlainTime,
+    serializeToUri: (value, serialize) => `time'${serialize(value)}'`
+  } as DeSerializer<Temporal.PlainTime>
+};
+
+/**
+ * @internal
+ */
+export function deserializeToPlainDateTime(
+  edmDateTime: string
+): Temporal.PlainDateTime {
+  const dateTimeOffsetComponents = parseEdmDateTimeOffset(edmDateTime);
+  const dateTimeInstant = Temporal.Instant.fromEpochMilliseconds(
+    parseInt(dateTimeOffsetComponents.ticks)
+  );
+  return Temporal.PlainDateTime.from(dateTimeInstant.toString());
+}
+
+/**
+ * @internal
+ */
+export function deserializeToZonedDateTime(
+  edmDateTime: string
+): Temporal.ZonedDateTime {
+  const dateTimeOffsetComponents = parseEdmDateTimeOffset(edmDateTime);
+  const dateTimeInstant = Temporal.Instant.fromEpochMilliseconds(
+    parseInt(dateTimeOffsetComponents.ticks)
+  );
+  if (dateTimeOffsetComponents.sign && dateTimeOffsetComponents.offset) {
+    const timeZone = Temporal.TimeZone.from(
+      dateTimeOffsetComponents.sign + dateTimeOffsetComponents.offset
+    );
+    return dateTimeInstant.toZonedDateTimeISO(timeZone);
+  }
+  return dateTimeInstant.toZonedDateTimeISO('UTC');
+}
+
+function parseEdmDateTimeOffset(edmDateTime: string): {
+  [key: string]: string;
+} {
```

Comment:
I would prefer a stronger typed return type since you use in the `deserializeToZonedDateTime` the ticks and offset and sign property.